### PR TITLE
Hide UserData field from the EditVM view for VMs that do not offer it

### PR DESCRIPTION
**Pull request by FelipeM525**
### Description
 
Currently in ACS, when editing a VM, the field userdata will be present even if the VM's network does not support the feature. This PR fixes this issue by not showing the field if the VM does not offer userdata.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### How Has This Been Tested?

I tested this by verifying if the userdata field would still show up in the Edit VM View when editing VMs whose networks did not offer userdata, and it worked as expected. I've also tested cases in which a VM could have more than one network, and for these cases it will consider the default network and disregard the other networks.

### CODE REVIEW

**Comment by kiranchavala** [comment]
@blueorangutan ui

**Review by kiranchavala — APPROVED** [review]
LGTM

Before fix

The edit vm  action had the userdata text box even if the network offering didn't support userdata

Screenshot 2024-09-26 at 11 34 49 AM

--
After fix 

There is no userdata box when user clicks on edit vm if the network offering didn't support userdata

Screenshot 2024-09-26 at 11 34 17 AM

File: ui/src/views/compute/EditVM.vue [hunk 1]
```diff
@@ -289,15 +290,35 @@ export default {
...
+      this.resource.nic.forEach(nic => {
+        if (nic.isdefault) {
+          networkId = nic.networkid
+        }
+      })
+      if (!networkId) {
+        return
+      }
+      const listNetworkParams = {
+        id: networkId
```
Comment (weizhouapache) [inline ui/src/views/compute/EditVM.vue:303]: Listall : true ?

File: ui/src/views/compute/EditVM.vue [hunk 2]
```diff
@@ -289,15 +290,35 @@ export default {
...
+      api(`listNetworks`, listNetworkParams).then(json => {
+        json.listnetworksresponse.network[0].service.forEach(service => {
+          if (service.name === 'UserData') {
+            this.userDataEnabled = true
 
-      api('listVirtualMachines', params).then(json => {
-        this.form.userdata = this.decodeUserData(json.listvirtualmachinesresponse.virtualmachine[0].userdata || '')
+            const listVmParams = {
+              id: this.resource.id,
+              userdata: true
```
Comment (weizhouapache) [inline ui/src/views/compute/EditVM.vue:312]: listall: true as well ?

File: ui/src/views/compute/EditVM.vue (hunk 1 above)
Comment (FelipeM525) [reply ui/src/views/compute/EditVM.vue:303]: Hello @weizhouapache , I did not quite understand your question; do you want me to add a `listall: true` as a parameter in the request? Could you give me some details on why this is needed?

Comment (weizhouapache) [reply ui/src/views/compute/EditVM.vue:303]: Hi @FelipeM525 
if the caller (for example root admin) is different as the network owner (for example regular user), then `listall=true` is needed


see an example below
```
(testadmin) 🐱 > list networks id=69a230d7-1534-4cc9-b791-b98fdc2a845e filter=name
(testadmin) 🐱 > 
(testadmin) 🐱 > 
(testadmin) 🐱 > list networks id=69a230d7-1534-4cc9-b791-b98fdc2a845e filter=name listall=true
{
  "count": 1,
  "network": [
    {
      "name": "testadmin-003"
    }
  ]
}
```

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Review by JoaoJandre — APPROVED** [review]
LGTM

**Comment by JoaoJandre** [comment]
Merging based on approvals and manual test by @kiranchavala (https://github.com/apache/cloudstack/pull/9731#pullrequestreview-2330202785)

**Pull request merged**